clients/src/main/java/io/strimzi/http/producer/HttpProducerClient.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.http.producer;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.grpc.netty.shaded.io.netty.handler.codec.http.HttpResponseStatus;
+import io.strimzi.common.ClientsInterface;
+import io.strimzi.common.configuration.Constants;
+import io.strimzi.common.configuration.http.HttpProducerConfiguration;
+import io.strimzi.test.tracing.HttpContext;
+import io.strimzi.test.tracing.HttpHandle;
+import io.strimzi.test.tracing.TracingHandle;
+import io.strimzi.test.tracing.TracingUtil;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.net.http.HttpClient;
+import java.net.http.HttpResponse;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class HttpProducerClient implements ClientsInterface {
+    private static final Logger LOGGER = LogManager.getLogger(HttpProducerClient.class);
+    private final HttpProducerConfiguration configuration;
+    private int messageIndex;
+    private int messageSuccessfullySent;
+    private HttpClient client;
+    private TracingHandle tracingHandle;
+    private HttpHandle httpHandle;
+    private final ScheduledExecutorService scheduledExecutor;
+    private CountDownLatch latch = new CountDownLatch(1);
+
+    public HttpProducerClient() {
+        this.configuration = new HttpProducerConfiguration(System.getenv());
+        this.messageIndex = 0;
+        this.client = HttpClient.newHttpClient();
+        this.tracingHandle = TracingUtil.getTracing();
+        this.httpHandle = tracingHandle.createHttpHandle("send-messages");
+        this.scheduledExecutor = Executors.newScheduledThreadPool(1, r -> new Thread(r, "http-producer"));
+    }
+
+    @Override
+    public void run() {
+        LOGGER.info("Starting {} with configuration: \n{}", this.getClass().getName(), configuration.toString());
+
+        scheduledExecutor.scheduleAtFixedRate(this::sendMessages, 0,  configuration.getDelay(), TimeUnit.MILLISECONDS);
+        awaitCompletion();
+    }
+
+    @Override
+    public void awaitCompletion() {
+        try {
+            latch.await();
+            scheduledExecutor.awaitTermination(Constants.DEFAULT_TASK_COMPLETION_TIMEOUT, TimeUnit.MILLISECONDS);
```

Comment:
what's the purpose of the latch? Isn't the `scheduledExecutor.awaitTermination` enough?

Reply:
IIRC it was because of "blocking of the actual thread", so it will wait until the task(s) will be completed and then just wait for executor termination - at least that's what I found that is "good practice"

Reply:
Tbh I still don't understand. The two await are anyway blocking the actual thread. Where did you find this "good practice" with these two blocking waits?

Reply:
The `awaitTermination` is anyway waiting for executor to be terminated after the shutdown, or? When here will be some default timeout, then the `awaitTermination` will throw waiting exception when we will send f.e. 1000 messages with 20s delay between each other or?

> Where did you find this "good practice" with these two blocking waits?
SO I think 🤷

Reply:
Anyway, I can try it without the `CountDownLatch` :)

Reply:
I have something similar in the client example and it works without the latch.
The trick is the timeout on the executor await. I think that test clients were already using something like:

num. messages * delay + <some jitter>

We would expect sending to ends within num.messages * delay, but we add a jitter just in case.

Reply:
Yeah I think that we are having something like that.
So yeah, I will change it :) thanks